### PR TITLE
New PhotStat index

### PR DESCRIPTION
**Pull request by guynir42**
Replace the PhotStat index on `mean_color` with a new index using GIN (generalized inverted index). 
This solves the issue that some objects with lots of filters with long names cannot be indexed on `mean_color` because their JSONB tag values are too long.

### CODE REVIEW

File: skyportal/models/phot_stat.py [hunk 1]
```diff
@@ -566,42 +574,71 @@ def full_update(self, phot_list):
...
+            mjds.append(phot['mjd'])
+
+            if 'mag' in phot:
+                mags.append(phot['mag'])
+            elif phot['flux'] is not None and phot['flux'] > 0:
+                mags.append(-2.5 * np.log10(phot['flux']) + PHOT_ZP)
+            else:
+                mags.append(np.nan)
+
+            is_detected = (
```
Comment (mcoughlin) [inline skyportal/models/phot_stat.py:611]: Can't the presence of magnitude also indicate detection?

**Review by mcoughlin — COMMENTED** [review]
@guynir42 only thing I wanted to check on is whether need to also check for magnitudes when checking for detections

File: skyportal/models/phot_stat.py (hunk 1 above)
Comment (guynir42) [reply skyportal/models/phot_stat.py:611]: I think you can have a 2 sigma detection with magnitude that doesn't count as a "detection".

Comment (mcoughlin) [reply skyportal/models/phot_stat.py:611]: @guynir42 reasonable! Ok so to me should be ok

**Pull request merged**